### PR TITLE
Adding Conduct, Contribution and Issues template to fit community practices

### PR DESCRIPTION
**Pull request by skahler-pivotal**
Reviewing best practices for Conduct, Contribution docs for the community and adding copies of those.

Also adding Issues template to try to help make issue reporting be more fluid

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@skahler-pivotal Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=3687)!

[Click here](https://cla.pivotal.io/sync/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=3687) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@skahler-pivotal Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=3687)!

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+## Coding guidelines
+
+Your chances of getting feedback and seeing your code merged into the project greatly depend on how granular your changes are. If you happen to have a bigger change in mind, we highly recommend engaging on the developer's mailing list first and sharing your proposal with us before you spend a lot of time writing code. Even when your proposal gets validated by the community, we still recommend doing the actual work as a series of small, self-contained commits. This makes the reviewer's job much easier and increases the timeliness of feedback.
+
+When it comes to C and C++ parts of Greenplum, we try to follow PostgreSQL Coding Conventions. In addition to that we require that:
+
+All Python code passes Pylint
+All Go code is formatted according to gofmt
+We recommend using git diff --color when reviewing your changes so that you don't have any spurious whitespace issues in the code that you submit.
```
Comment (danielgustafsson) [inline .github/CONTRIBUTING.md]: Wouldn't it be better to suggest `git diff --check` to handle whitespace issues?

File: .github/issue_template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+### Greenplum version or build
```
Comment (danielgustafsson) [inline .github/issue_template.md]: If we add the command we expect the user to run to gather this information we will likely get better reports. Also we should add that we want to see operating system and configuration options (the autoconf options used).

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (skahler-pivotal) [reply .github/CONTRIBUTING.md]: It looks to me like --check would help to specifically find whitespace, but color will help notice whitespace issues in a standard git diff review. I think the intention there is not to check specifically for whitespaces, but to make sure color is there during a normal review to help highlight that type of issue.

**Comment by skahler-pivotal** [comment]
We probably could automatically scrape to greenplum.org, will need to look into that. I would suggest I drop a note on greenplum.org noting that the information there is pulled from here and include the date of the last pull for now.

**Comment by danielgustafsson** [comment]
> On 31 Oct 2017, at 14:41, Scott Kahler <notifications@github.com> wrote:
> 
> We probably could automatically scrape to greenplum.org, will need to look into that. I would suggest I drop a note on greenplum.org noting that the information there is pulled from here and include the date of the last pull for now.

Sounds like a good start

File: .github/issue_template.md (hunk 2 above)
Comment (skahler-pivotal) [reply .github/issue_template.md]: added some of the suggested specifics on things to run to help with issue environment collection

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (amilkh) [reply .github/CONTRIBUTING.md]: This README was merged just today, but might be worth adding a note here like "pgindent is applied on code changes as described here"
https://github.com/greenplum-db/gpdb/blob/master/src/tools/pgindent/README.gpdb

**Pull request merged**